edge/pkg/devicetwin/dtmanager/device.go

```diff
@@ -159,6 +160,9 @@ func UpdateDeviceAttr(context *dtcontext.DTContext, deviceID string, attributes
 		return nil, nil
 	}
 	dealAttrResult := DealMsgAttr(context, Device.ID, attributes, dealType)
+	if dealAttrResult.Err != nil {
+		return nil, err
+	}
```

Comment:
seems reasonable, if err occur, Add, Delete, Update field will be nil.

Reply:
better `return nil, nil`
the original logic is always returned `nil, nil`. 😄

Reply:
already edited